frontend/src/app/store/reducers/complaints.ts

```diff
@@ -105,6 +105,10 @@ export const complaintSlice = createSlice({
 
     setComplaint: (state, action) => {
       const { payload: complaint } = action;
+      let currentComplaint: ComplaintDto = complaint as ComplaintDto;
+      if (currentComplaint) {
+        currentComplaint.readOnly = ["CLOSED"].includes(currentComplaint.status);
```

Comment:
Instead of adding a pseudo-property on the data object itself, could we follow the same pattern as the "isInEdit" flag that controls the edit behaviour and create an equivalent isReadOnly flag? You could dispatch a "setIsReadOnly" action here instead. The overall logic would remain the same but this is consistent with an existing pattern.